File: C++/523. Continuous Subarray Sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        unordered_map<int, int>maple;
        int sum = 0;
        maple[0] = -1;
        for(int i=0;i<nums.size();i++)
        {
            sum += nums[i];
            if(k) sum %= k;
            if(maple.find(sum) != maple.end())
            {
                if(i-maple[sum]>1)
                    return true;
            }
            else
                maple[sum] = i;
        }
        return false;
    }
};

File: Python/523. Continuous Subarray Sum.py
class Solution:
    def checkSubarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        maple = {0:-1}
        res = 0
        for i in range(len(nums)):
            res += nums[i]
            if k != 0:
                res %= k
            index = maple.get(res)
            if index != None:
                if i - index > 1:
                    return True
            else:
                maple[res] = i
        return False